lib/pbench/test/unit/common/test_utils.py

```diff
@@ -18,3 +20,47 @@ def test_md5sum():
         assert (
             hash_md5 == expected_hash_md5
         ), f"Expected MD5 '{expected_hash_md5}', got '{hash_md5}'"
+
+
+class TestCleanup:
+    def test_contruct(self, caplog):
+        logger = logging.getLogger("test_construct")
+        c = Cleanup(logger)
+        assert isinstance(c.actions, collections.deque)
+        assert len(c.actions) == 0
+        assert c.logger is logger
+
+    def test_add(self, caplog):
+        class Test:
+            def delete(self):
+                pass
+
+        logger = logging.getLogger("test_construct")
+        c = Cleanup(logger)
+        t = Test()
+        c.add(t.delete)
+        assert c.actions.pop().action == t.delete
```

Comment:
This test is sort of borderline -- again, it's peeking inside the unit, and I'm not sure that that is a good idea.  I think that `test_cleanup()` covers the cases that this test covers, so I think this test is superfluous.